src/looprestoration.rs

```diff
@@ -934,3 +955,194 @@ unsafe fn sgr_mix_rust<BD: BitDepth>(
         j += 1;
     }
 }
+
+// TODO(randomPoison): Temporarily pub until all usages can be made private.
+#[cfg(all(feature = "asm", any(target_arch = "arm", target_arch = "aarch64")))]
+pub(crate) trait BitDepthLooprestorationArm: BitDepth {
+    fn dav1d_sgr_box3_h_neon(
+        sumsq: *mut int32_t,
+        sum: *mut int16_t,
+        left: *const [Self::Pixel; 4],
+        src: *const Self::Pixel,
+        stride: ptrdiff_t,
+        w: libc::c_int,
+        h: libc::c_int,
+        edges: LrEdgeFlags,
+    );
+
+    fn dav1d_sgr_finish_filter1_neon(
+        tmp: *mut int16_t,
+        src: *const Self::Pixel,
+        stride: ptrdiff_t,
+        a: *const int32_t,
+        b: *const int16_t,
+        w: libc::c_int,
+        h: libc::c_int,
+    );
+}
+
+#[cfg(all(feature = "asm", any(target_arch = "arm", target_arch = "aarch64")))]
+impl BitDepthLooprestorationArm for BitDepth8 {
+    #[inline(always)]
+    fn dav1d_sgr_box3_h_neon(
+        sumsq: *mut int32_t,
+        sum: *mut int16_t,
+        left: *const [Self::Pixel; 4],
+        src: *const Self::Pixel,
+        stride: ptrdiff_t,
+        w: libc::c_int,
+        h: libc::c_int,
+        edges: LrEdgeFlags,
+    ) {
+        extern "C" {
+            fn dav1d_sgr_box3_h_8bpc_neon(
+                sumsq: *mut int32_t,
+                sum: *mut int16_t,
+                left: *const [<BitDepth8 as BitDepth>::Pixel; 4],
+                src: *const <BitDepth8 as BitDepth>::Pixel,
+                stride: ptrdiff_t,
+                w: libc::c_int,
+                h: libc::c_int,
+                edges: LrEdgeFlags,
+            );
+        }
+
+        unsafe { dav1d_sgr_box3_h_8bpc_neon(sumsq, sum, left, src, stride, w, h, edges) }
```

Comment:
We should propagate the `unsafe` here.  I've been putting safety comments on all `unsafe {}`s.

Reply:
Switching to using associated consts also addresses this 👍